ndcube/ndcube.py

```diff
@@ -409,21 +419,46 @@ def extra_coords(self):
                     "value": self._extra_coords_wcs_axis[key]["value"]}
         return result
 
-    def crop_by_coords(self, min_coord_values, interval_widths):
+    def crop_by_coords(self, lower_corner, interval_widths=None, upper_corner=None):
         # The docstring is defined in NDDataBase
 
         n_dim = len(self.dimensions)
```

Comment:
I know this wasn't your change, but can you replace this is ```n_dim = self.data.ndim```?  It's more efficient.

Reply:
Changed